src/history/hs_rec.c

```diff
@@ -213,10 +219,26 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
         error_on_ts_ordering = false;
     }
 
-    if (ret == 0)
-        WT_ERR(__hs_delete_reinsert_from_pos(session, cursor, btree->id, key, tw->start_ts + 1,
-          true, false, error_on_ts_ordering, &counter, tw));
-
+    if (ret == 0) {
+        WT_ASSERT(session, tw->start_ts + 1 > WT_TS_NONE);
+        /* Get the history store cursor position. */
+        WT_ERR_NOTFOUND_OK(__hs_cursor_pos(session, cursor, btree->id, key, tw->start_ts + 1,
+                             error_on_ts_ordering, tw, &non_ts_updates),
+          true);
+
+        if (ret == 0) {
+            /* For non-timestamped updates we only need the counter, skip reinsert. */
+            if (non_ts_updates) {
```

Comment:
no need for brankets.

Reply:
removed here [1c4f7bb](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/1c4f7bb287446e1166c6d19da2501a9c365bff50)